### ISSUE TITLE
iOS Auto-layout/Orientation/split-screen not working using 46 and dev-client

### ISSUE THREAD

**Issue by chrisdrackett** [expo/expo#18689]
### Summary

I'm running into an issue where when running an app the size is stuck to the size of the device when the app is loaded. If the screen changes in any of the following ways:

- rotate device
- resize (mac)
- split screen (ipad)

The app still renders at the initial size and parts of the app may not be accessible. Also areas below or beside the app at its initial size will be shown as black/gray space.

What I find interesting is that the "home" view of expo dev (where it lists serves, extensions and settings) is not affected and resizes correctly in all the above case. It's only once you load up the client app that this issue happens.

The related settings I have in `app.json` are:

```
    "ios": {
      "supportsTablet": true,
      "requireFullScreen": false,
     }
```

This bug is similar to #14967 at least in how it presents itself to the user.

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

iOS

### Package versions

`expo@npm:46.0.7`
`react-native@npm:0.69.4`
`expo-dev-client@nppm:1.2.1`
`expo-updates@npm:0.14.4`

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.0 - /usr/local/bin/node
      Yarn: 3.2.0 - /usr/local/bin/yarn
      npm: 8.1.0 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    Expo Workflow: managed

### Reproducible demo

This happens for me with the base `npx create-expo-app` project

**Comment by esamelson** [comment]
Hey @chrisdrackett -- thanks for the report! Is there a specific iOS version / device / simulator type that this happens on? I tried reproducing it on a couple different simulators and couldn't -- here is iPad mini simulator running iOS 15.2

https://user-images.githubusercontent.com/19958240/186042653-36b162f8-fe10-4640-a695-f219fc8d678e.mp4

Also wanted to double check that after upgrading to `expo-dev-client@1.2.1` you made a new build? Asking since a very similar issue was fixed in 1.2.0 (https://github.com/expo/expo/pull/18509) that would require a new build to take effect.
